locking/locks.go

```diff
@@ -47,15 +50,20 @@ func NewClient(cfg *config.Configuration) (*Client, error) {
 	return &Client{cfg, apiClient, store}, nil
 }
 
+// Close this client instance; must be called to dispose of resources
+func (c *Client) Close() error {
+	return c.cache.Save()
+}
+
 // LockFile attempts to lock a file on the current remote
 // path must be relative to the root of the repository
 // Returns the lock id if successful, or an error
-func (c *Client) LockFile(path string) (id string, e error) {
+func (c *Client) LockFile(path string) (Lock, error) {
```

Comment:
That depends on callers doing the right thing (not modifying return values from the locking client. I think cache corruption is a good enough reason to return a value.